tools/gbench/report.py

```diff
@@ -142,33 +142,41 @@ def extract_field(partition, field_name):
     rhs = [x[field_name] for x in partition[1]]
     return [lhs, rhs]
 
-
-def print_utest(partition, utest_alpha, first_col_width, use_color=True):
-    timings_time = extract_field(partition, 'real_time')
-    timings_cpu = extract_field(partition, 'cpu_time')
-
+def calc_utest(timings_cpu, timings_time):
     min_rep_cnt = min(len(timings_time[0]),
                       len(timings_time[1]),
                       len(timings_cpu[0]),
                       len(timings_cpu[1]))
 
     # Does *everything* has at least UTEST_MIN_REPETITIONS repetitions?
     if min_rep_cnt < UTEST_MIN_REPETITIONS:
-        return []
-
-    def get_utest_color(pval):
-        return BC_FAIL if pval >= utest_alpha else BC_OKGREEN
+        return False, None, None
 
     time_pvalue = mannwhitneyu(
         timings_time[0], timings_time[1], alternative='two-sided').pvalue
     cpu_pvalue = mannwhitneyu(
         timings_cpu[0], timings_cpu[1], alternative='two-sided').pvalue
 
+    return (min_rep_cnt >= UTEST_OPTIMAL_REPETITIONS), cpu_pvalue, time_pvalue
+
+def print_utest(partition, utest_alpha, first_col_width, use_color=True):
+    def get_utest_color(pval):
+        return BC_FAIL if pval >= utest_alpha else BC_OKGREEN
+
+    timings_time = extract_field(partition, 'real_time')
+    timings_cpu = extract_field(partition, 'cpu_time')
+    optimal_repetitions, cpu_pvalue, time_pvalue = calc_utest(timings_cpu, timings_time)
```

Comment:
`timings_time` and `timings_cpu` are not used in this function, only the `len(timings_cpu[0]), len(timings_cpu[1])`.
How about you pass `partition` into the `calc_utest()`, and make it additionally return those lenghts?

Reply:
Hmmm, I'm not sure about this one. `calc_utest()`, as it's name implies, return a u-test result no less no more. Narrowing the scope of the functions usually results in less bugs because there are less things that you can do wrong. So, tbh, I am not sure if this change would improve on the code.

Reply:
I believe that this change now became sort of obsolete (or irrelevant) because IMO is that now we have much better solution with changes being done in #737. See 9021718 to get a feeling how it looks like.